code/game/turfs/simulated/floor.dm

```diff
@@ -103,496 +103,68 @@ var/list/wood_icons = list("wood","wood-broken")
-//If no argument is given a new one will be made.
-/turf/simulated/floor/proc/make_grass_floor(var/obj/item/stack/tile/grass/T = null)
-	broken = 0
-	burnt = 0
-	intact = 1
-	if(T)
-		if(istype(T,/obj/item/stack/tile/grass))
-			floor_tile = T
-			update_icon()
-			levelupdate()
-			return
-	//if you gave a valid parameter, it won't get thisf ar.
-	floor_tile = new/obj/item/stack/tile/grass
-
-	update_icon()
-	levelupdate()
-
-//This proc will make a turf into a wood floor. Fun eh? Insert the wood tile to be used as the argument
-//If no argument is given a new one will be made.
-/turf/simulated/floor/proc/make_wood_floor(var/obj/item/stack/tile/wood/T = null)
-	broken = 0
-	burnt = 0
-	intact = 1
-	if(T)
-		if(istype(T,/obj/item/stack/tile/wood))
-			floor_tile = T
-			update_icon()
-			levelupdate()
-			return
-	//if you gave a valid parameter, it won't get thisf ar.
-	floor_tile = new/obj/item/stack/tile/wood
-
-	update_icon()
-	levelupdate()
-
-//This proc will make a turf into a carpet floor. Fun eh? Insert the carpet tile to be used as the argument
-//If no argument is given a new one will be made.
-/turf/simulated/floor/proc/make_carpet_floor(var/obj/item/stack/tile/carpet/T = null)
-	broken = 0
-	burnt = 0
-	intact = 1
-	if(T)
-		if(istype(T,/obj/item/stack/tile/carpet))
-			floor_tile = T
-			update_icon()
-			levelupdate()
-			return
-	//if you gave a valid parameter, it won't get thisf ar.
-	floor_tile = new/obj/item/stack/tile/carpet
-
-	update_icon()
-	levelupdate()
 
 /turf/simulated/floor/attackby(obj/item/C as obj, mob/user as mob, params)
 	if(!C || !user)
-		return 0
-
-	if(istype(C,/obj/item/weapon/light/bulb)) //only for light tiles
-		if(is_light_floor())
-			var/obj/item/stack/tile/light/T = floor_tile
-			if(T.state)
-				user.drop_item(C)
-				qdel(C)
-				T.state = C //fixing it by bashing it with a light bulb, fun eh?
-				update_icon()
-				user << "<span class='notice'>You replace the light bulb.</span>"
-			else
-				user << "<span class='notice'>The light bulb seems fine, no need to replace it.</span>"
-
-	if(istype(C, /obj/item/weapon/crowbar) && (!(is_plating())) && (!is_catwalk()))
+		return 1
+	if(..())
+		return 1
+	if(intact && istype(C, /obj/item/weapon/crowbar))
 		if(broken || burnt)
-			user << "<span class='notice'>You remove the broken plating.</span>"
+			broken = 0
+			burnt = 0
+			user << "<span class='danger'>You remove the broken plating.</span>"
 		else
-			if(is_wood_floor())
+			if(istype(src, /turf/simulated/floor/wood))
 				user << "<span class='danger'>You forcefully pry off the planks, destroying them in the process.</span>"
 			else
-				user << "<span class='notice'>You remove the [floor_tile.name].</span>"
-				new floor_tile.type(src)
-
+				user << "<span class='danger'>You remove the floor tile.</span>"
+				builtin_tile.loc = src
+				builtin_tile = null //deassociate tile, it no longer belongs to this turf

```

Comment:
I actually have no idea how -tg- currently functions, I presume that turfs made into plating do not call Destroy(), as the Destroy() proc deletes builtin tiles.